### PR TITLE
Use polling for login and room joining

### PR DESCRIPTION
**Pull request by msujew**
Closes https://github.com/TypeFox/open-collaboration-tools/issues/35
Attempts to fix https://github.com/TypeFox/open-collaboration-tools/issues/34

Uses a different networking solution when logging in and joining rooms. Instead of waiting until either action is completed until returning a result, both now are using a polling approach. This has two benefits:
1. We can use the polling of the client to send updates about the process to the user (e.g. whether the join request is currently awaiting input from the host).
2. It is less affected by a few bad requests. Since we regularly resend the polling requests, we can always ask again whether the login or join request has succeeded.

This is a breaking change in the server communication protocol. After merging this, we should update the protocol version to `0.2.0`.

### CODE REVIEW

File: packages/open-collaboration-server/src/credentials-manager.ts [hunk 1]
```diff
@@ -38,18 +40,22 @@ export class CredentialsManager {
         if (this.configuration.getValue('oct-jwt-private-key') === undefined) {
             this.logger.warn('OCT_JWT_PRIVATE_KEY env variable is not set. Using a static key for development purposes.');
         }
+        if (this.configuration.getValue('oct-rsa-public-key') === undefined) {
```
Comment (jonah-iden) [inline packages/open-collaboration-server/src/credentials-manager.ts:43]: just out of curiosity, i don't quite get what the server needs a key pair for? I thought the encryption is just between the peers.

**Review by jonah-iden — APPROVED** [review]
Works like a charm! Code looks good to me

File: packages/open-collaboration-server/src/credentials-manager.ts (hunk 1 above)
Comment (jonah-iden) [reply packages/open-collaboration-server/src/credentials-manager.ts]: Thanks for the explanation in your latest commit! Looks good to me

**Review by spoenemann — COMMENTED** [review]
I see the new message "Waiting for host to accept join request". However, when I click "Allow" as the host, the guest does not join the room. I didn't see any error message.

**Comment by msujew** [comment]
@spoenemann I cannot reproduce. How did you test the feature? Did you use `vsce package --no-dependencies` to build the extension and install that in a second vscode instance?

**Pull request merged**